huaweicloud/services/iec/resource_huaweicloud_iec_vip.go

```diff
@@ -101,87 +106,95 @@ func resourceIecVIPV1Create(d *schema.ResourceData, meta interface{}) error {
 
 	p, err := ports.Create(iecClient, createOpts).Extract()
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud IEC port: %s", err)
+		return diag.Errorf("error creating IEC port: %s", err)
 	}
 
-	logp.Printf("[DEBUG] Waiting for HuaweiCloud IEC Port (%s) to become available.", p.ID)
+	log.Printf("[DEBUG] Waiting for IEC Port (%s) to become available.", p.ID)
 	stateConf := &resource.StateChangeConf{
 		Target:     []string{"ACTIVE"},
 		Refresh:    waitingForIECVIPActive(iecClient, p.ID),
 		Timeout:    d.Timeout(schema.TimeoutCreate),
 		Delay:      5 * time.Second,
 		MinTimeout: 3 * time.Second,
 	}
-	_, err = stateConf.WaitForState()
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error creating IEC port: %s", err)
```

Comment:
the error message is not correct

```suggestion
		return diag.Errorf("error waiting for the IEC port (%s) to become available: %s", err)
```